context/handlers.go

```diff
@@ -16,3 +34,12 @@ func TenantIDHandler(next http.Handler) http.Handler {
 		next.ServeHTTP(w, r.WithContext(ContextWithTenantID(r.Context(), tenantID)))
 	})
 }
+
+func MaxConnectionAgeHandler(next http.Handler, maxConnectionAge time.Duration) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		if start, ok := ConnectionStartTimeFromContext(r.Context()); ok && clock.Now().After(start.Add(maxConnectionAge)) {
+			w.Header().Add("Connection", "close")
```

Comment:
**suggestion:** I believe you should use `Set` instead of `Add`. `Add` will include any existing `Connection` header in the response.

```suggestion
			w.Header().Set("Connection", "close")
```

Reply:
Good catch thanks!